domain-mapping/api/class-darkmatter-domains.php

```diff
@@ -360,6 +410,108 @@ public function get( $fqdn = '' ) {
 		return new DM_Domain( (object) $_domain );
 	}
 
+	/**
+	 * Retrieve a List of domains by type. For example: retrieving a list of Media domains.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param  integer $type       Domain type to retrieve.
+	 * @param  integer $site_id    Site ID to retrieve mapped domains for.
+	 * @param  boolean $skip_cache Skip the cache and retrieve the Media domains from the database.
+	 * @return array               An array of DM_Domain objects. Returns an empty array if no mapped domains found or on error.
+	 */
+	public function get_domains_by_type( $type = DM_DOMAIN_TYPE_MEDIA, $site_id = 0, $skip_cache = false ) {
+		global $wpdb;
+
+		/**
+		 * Validate type before continuing.
+		 */
+		if ( DM_DOMAIN_TYPE_MAIN !== $type && DM_DOMAIN_TYPE_MEDIA !== $type ) {
+			return [];
+		}
+
+		/**
+		 * Media domains can be set with a constant.
+		 */
+		if ( DM_DOMAIN_TYPE_MEDIA === $type && defined( 'DM_NETWORK_MEDIA' ) ) {
+			/**
+			 * Ensure it is an array and then return. No checks, we assume that is provided correctly.
+			 */
+			if ( is_array( DM_NETWORK_MEDIA ) ) {
+				$media_domains = [];
+
+				/**
+				 * Convert the domains into DM_Domain objects.
+				 */
+				foreach ( DM_NETWORK_MEDIA as $i => $media_domain ) {
+					$media_domains[ $i ] = new DM_Domain( (object) [
+						'active'     => true,
+						'blog_id'    => get_current_blog_id(),
+						'domain'     => $media_domain,
+						'id'         => -1,
+						'is_https'   => true,
+						'is_primary' => false,
+						'type'       => DM_DOMAIN_TYPE_MEDIA,
+					] );
+				}
+
+				return $media_domains;
+			}
+		}
+
+		/**
+		 * Setup the cache key.
+		 */
+		$site_id   = ( empty( $site_id ) ? get_current_blog_id() : $site_id );
+		$cache_key = md5(
+			sprintf(
+				'%1$d-%2$d-domain-types',
+				$type,
+				$site_id
+			)
+		);
```

Comment:
can call the refactored get_cache_key function here should you go ahead with it, otherwise it's a duplicate code fragment.